### PR TITLE
adding expiration notification email to check-user-expiry-lambda

### PR DESCRIPTION
**Pull request by MattToczek**

### CODE REVIEW

File: terraform/modules/check-user-expiry-lambda/lambda.tf [hunk 1]
```diff
@@ -8,11 +8,12 @@ resource "aws_lambda_function" "lambda_check_user_expiry" {
...
-      TABLE_NAME           = var.dynamodb_table_user_name,
-      BUCKET_NAME          = var.template_bucket,
-      COGNITO_USER_POOL_ID = var.cognito_user_pool_id,
-      MAIL_FROM            = var.from_email_address,
-      SUBJECT_LINE         = "[[ recipient_name ]] Your access to DataWorks AWS is about to expire"
+      TABLE_NAME            = var.dynamodb_table_user_name,
+      BUCKET_NAME           = var.template_bucket,
+      COGNITO_USER_POOL_ID  = var.cognito_user_pool_id,
+      MAIL_FROM             = var.from_email_address,
+      REMINDER_SUBJECT_LINE = "[[ recipient_name ]] Your access to DataWorks AWS is about to expire"
```
Comment (IanUtley) [inline terraform/modules/check-user-expiry-lambda/lambda.tf:15]: Should there not be full stops after the recipient name?

Comment (MattToczek) [reply terraform/modules/check-user-expiry-lambda/lambda.tf:15]: I will add them - I just copied the existing code

File: terraform/modules/check-user-expiry-lambda/lambda/lambda_function.py [hunk 2]
```diff
@@ -66,11 +72,22 @@ def query_dynamodb_users_about_expire():
     return response["Items"]
 
 
-def process_items(items):
+def query_dynamodb_users_expired_today():
+    print("Query dynamodb table user for users expiring today")
+    today = datetime.date.today()
+    response = table.scan(
+        FilterExpression=Attr("expiration_date").eq(
```
Comment (IanUtley) [inline terraform/modules/check-user-expiry-lambda/lambda/lambda_function.py:79]: Looking at expiration dates in the DB, they are datetimes, so will they ever be equal to 'today'? Does str(today) display it in the same ISO8601 format as its dynamo stores dates as Strings?

File: terraform/modules/check-user-expiry-lambda/lambda/lambda_function.py [hunk 3]
```diff
@@ -109,19 +126,25 @@ def extract_email_from_user_attributes(user):
...
-    if items:
-        print("Query returned " + str(len(items)) + " item(s)")
-        process_items(items)
+    about_to_expire = query_dynamodb_users_about_expire()
+    if about_to_expire:
+        print("Query returned " + str(len(about_to_expire)) + " item(s)")
+        process_items(about_to_expire, reminder_subject_line, "default_email_template_analytical.html")
+    else:
+        print("Query did not return any due to expire items.")
+    expired = query_dynamodb_users_about_expire()
```
Comment (IanUtley) [inline terraform/modules/check-user-expiry-lambda/lambda/lambda_function.py:145]: Arent you calling the wrong function here, shouldnt it be query_dynamodb_users_expired_today?

**Pull request merged**